grails-app/services/org/pih/warehouse/inventory/StockMovementService.groovy

```diff
@@ -939,13 +941,15 @@ class StockMovementService {
                     where parent_requisition_item_id in (${editItemsIds})
                     """).groupBy { it.parent_requisition_item_id }
 
-        def products = Product.findAllByIdInList(data.collect { it.product_id })
-        def productsMap = products.inject([:]) { map, item -> map << [(item.id): item] }
+        List<Product> products = Product.findAllByIdInList(data.collect { it.product_id })
+        Map productsMap = products.inject([:]) { map, item -> map << [(item.id): item] }
 
         Requisition requisition = Requisition.get(data.first()?.requisition_id)
         def picklistItemsMap = requisition?.picklist?.pickablePicklistItemsByProductId
 
-        def availableItemsMap = productAvailabilityService.getAllAvailableBinLocations(requisition.origin, products).groupBy { it?.inventoryItem?.product?.id }
+        Map<String, List<AvailableItem>> availableItemsMap = productAvailabilityService
+                .getAllAvailableBinLocations(requisition.origin, productsMap.keySet().stream().collect(Collectors.toList()))
```

Comment:
This was my initial suggestion. Later I suggested moving `data.collect { it.product_id }` from line 944 to separate variable and use it in `findAll...` and here.

Reply:
fixed